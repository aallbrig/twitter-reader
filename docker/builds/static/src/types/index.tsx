export type Media = {
    id: string,
    url: string
};
export type Hashtag = {
    tag: string
};
export type Entity = {
    hashtags: Hashtag[],
    media: Media[]
};
export type Tweet = {
    id: number,
    text: string,
    entities: Entity
};
export type FilterableTweet = Tweet & {
    disabled: boolean
};
export type GetRecentTweetsResponse = {
    status: 'OK',
    tweets: Tweet[]
};

export interface TweetState {
    tweets: Tweet[];
}
export interface FilterableTweetState {
    filterBy: string;
    filteredTweets: FilterableTweet[];
}
export interface StoreState {
    tweets: TweetState;
    filterTweets: FilterableTweetState;
}
