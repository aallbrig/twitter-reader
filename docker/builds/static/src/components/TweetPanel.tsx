import * as React from 'react';
import { Panel, Media, Row, Col } from 'react-bootstrap';
import { FormattedNumber, FormattedDate, FormattedTime } from 'react-intl';
import { FilterableTweet } from '../types';
const Highlight = require('react-highlighter')

export interface Props {
    tweet: FilterableTweet;
    index: number;
    highlightedWord: string;
}

export const TweetPanel: React.SFC<Props> = ({ tweet, index, highlightedWord }) => (
    <Panel
        {...(tweet.disabled ? {
            bsStyle: 'danger',
            // className: 'text-muted'
        } : {})}
        footer={tweet.entities.media.length > 0 && (
            <Row>
                <Col xs={12}>
                    <h4>Attached Media</h4>
                    {tweet.entities.media.map(({ id, url}) => (
                        <Media key={`media-${id}`}>
                            <a href={url} target="_blank">
                                <img
                                    className="thumbnail img-responsive"
                                    src={url}
                                    style={{maxHeight: 250}}
                                />
                            </a>
                        </Media>
                    ))}
                </Col>
            </Row>
        )}
    >
        <Row>
            <Col xs={3} sm={2}>
                <img
                    src={tweet.user.profile.default}
                    className="img-responsive thumbnail"
                />
                {tweet.user.name}
                <br />
                {`@${tweet.user.screen_name}`}
                <br />
                <br />
                {'Friends:'}
                <br className="visible-xs"/>
                {` `}
                <FormattedNumber value={tweet.user.friends_count} />
            </Col>
            <Col xs={7} style={{ wordWrap: 'break-word' }}>
                <h4>Tweet ID: {tweet.id}</h4>
                <div className="text-muted" style={{ marginBottom: 10 }}>
                    <FormattedTime value={tweet.created_at} />
                    {' '}
                    <FormattedDate
                        value={tweet.created_at}
                        year="numeric"
                        month="long"
                        day="2-digit"
                    />
                </div>
                <Highlight
                    search={highlightedWord}
                    matchElement="span"
                    matchClass="bg-danger"
                    matchStyle={{ border: '1px solid red' }}
                >
                    {tweet.text}
                </Highlight>
                {tweet.entities.hashtags.length > 0 && (
                    <div>
                        <hr />
                        <h5>
                            {'Hashtags: '}
                            {tweet.entities.hashtags
                                .map(({ tag }) => `#${tag}`).join(', ')}
                        </h5>
                    </div>
                )}
            </Col>
            <Col xs={2} sm={3}>
                <div className="pull-right text-right">
                    <p>
                        <span className="badge">{index}</span>
                    </p>
                    <p>
                        {'Favorite'}
                        <span className="hidden-xs">
                            {' count:'}
                        </span>
                        <br className="visible-xs"/>
                        {' '}
                        <i className="glyphicon glyphicon-heart" />
                        {` ${tweet.favorite_count}`}
                    </p>
                    <p>
                        {'Retweet'}
                        <span className="hidden-xs">
                            {' count:'}
                        </span>
                        <br className="visible-xs"/>
                        {' '}
                        <i className="glyphicon glyphicon-sunglasses" />
                        {` ${tweet.retweet_count}`}
                    </p>
                </div>
            </Col>
        </Row>
    </Panel>
);

export default TweetPanel;
