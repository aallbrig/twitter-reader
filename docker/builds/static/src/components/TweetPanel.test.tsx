import * as React from 'react';
import { IntlProvider } from 'react-intl';
import TweetPanel from './TweetPanel';
import { create } from 'react-test-renderer';
import { FilterableTweet } from '../types';

const mockTweet: FilterableTweet = {
    'id': 927523478357663700,
    'user': {
        'id': 33612317,
        'name': 'Salesforce',
        'screen_name': 'salesforce',
        'friends_count': 50032,
        'profile': {
            'mini': 'http://pbs.twimg.com/profile_images/716283264881700865/voT_NePC_mini.jpg',
            'normal': 'http://pbs.twimg.com/profile_images/716283264881700865/voT_NePC_normal.jpg',
            'bigger': 'http://pbs.twimg.com/profile_images/716283264881700865/voT_NePC_bigger.jpg',
            'default': 'http://pbs.twimg.com/profile_images/716283264881700865/voT_NePC.jpg'
        }
    },
    'text':
        `If coding makes you feelðŸ˜±, you'llðŸ˜Š this. NEW myEnstein` +
        ` lets you build AI-powered apps with just a few mouse clicks.â€¦ https://t.co/FdOPeF5G0j`,
    'created_at': '1509973804000',
    'entities': {
        'hashtags': [],
        'media': []
    },
    'favorite_count': 5,
    'retweet_count': 4,
    'coordinates': null,
    disabled: false
};

it('TweetPanel renders correctly', () => {
  const tree = create(
    <IntlProvider locale="en">
        <TweetPanel
            tweet={mockTweet}
            index={0}
            highlightedWord={''}
        />
    </IntlProvider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});