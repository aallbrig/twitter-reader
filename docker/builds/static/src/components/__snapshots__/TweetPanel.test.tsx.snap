// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TweetPanel renders correctly 1`] = `
<div
  className="panel panel-default"
  id={undefined}
  style={
    Object {
      "fontSize": "0.85em",
    }
  }
>
  <div
    className="panel-body"
  >
    <div
      className="row"
    >
      <div
        className="col-sm-2 col-xs-3"
      >
        <a
          href="http://pbs.twimg.com/profile_images/716283264881700865/voT_NePC.jpg"
          target="_blank"
        >
          <img
            className="img-responsive thumbnail"
            src="http://pbs.twimg.com/profile_images/716283264881700865/voT_NePC.jpg"
          />
        </a>
        Salesforce
        <br />
        @salesforce
        <br />
        <br />
        Friends:
        <br
          className="visible-xs"
        />
         
        <span>
          50,032
        </span>
      </div>
      <div
        className="col-xs-7"
        style={
          Object {
            "wordWrap": "break-word",
          }
        }
      >
        <h4>
          Tweet ID 
          <span
            className="text-info"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            927,523,478,357,663,700
          </span>
        </h4>
        <div
          className="text-muted"
          style={
            Object {
              "marginBottom": 10,
            }
          }
        >
          <span>
            Invalid Date
          </span>
           
          <span>
            Invalid Date
          </span>
        </div>
        <span>
          If coding makes you feelðŸ˜±, you'llðŸ˜Š this. NEW myEnstein lets you build AI-powered apps with just a few mouse clicks.â€¦ https://t.co/FdOPeF5G0j
        </span>
      </div>
      <div
        className="col-sm-3 col-xs-2"
      >
        <div
          className="pull-right text-right"
        >
          <p>
            <span
              className="badge"
              style={
                Object {
                  "backgroundColor": "initial",
                  "border": "1px solid #aaa",
                  "color": "#aaa",
                }
              }
            >
              0
            </span>
          </p>
          <p>
            Favorite
            <span
              className="hidden-xs"
            >
               count:
            </span>
            <br
              className="visible-xs"
            />
             
            <i
              className="glyphicon glyphicon-heart"
            />
             5
          </p>
          <p>
            Retweet
            <span
              className="hidden-xs"
            >
               count:
            </span>
            <br
              className="visible-xs"
            />
             
            <i
              className="glyphicon glyphicon-sunglasses"
            />
             4
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
